cg/meta/workflow/analysis.py

```diff
@@ -165,6 +161,19 @@ def get_case_application_type(self, case_id: str) -> str:
 
         return application_types.pop()
 
+    def get_case_source_type(self, case_id: str) -> str:
+        """Returns the source type for samples in a case.
+        Raises CgError if different sources are set for the samples linked to a case."""
+        sample_ids: Iterator[str] = self.status_db.get_sample_ids_by_case_id(case_id=case_id)
+        source_types: set[str] = {
+            self.lims_api.get_source(lims_id=sample_id) for sample_id in sample_ids
+        }
+
+        if len(source_types) > 1:
+            raise CgError(f"Different source types found for case: {case_id} ({source_types})")
```

Comment:
The function `self.lims_api.get_source` returns `None` if the sample with the lims id is not found. If this is the case, there will be more than one entry in the set `source_types`. Then the error message would be misleading. This behaviour should be handled explicitly. I'm not sure where, but maybe directly in `self.lims_api.get_source` so that it does not return None if the sample does not exist.

Reply:
updated `get_source` to handle directly the exception and return 'unknown' when no lims information can be fetched retrieved

Reply:
That's why `source_types` it's a `set`.  We'll have a single `None` there and the method would behave as expected. 

Just make sure to update the typehints of the method and source types

Reply:
discussed in huddle. The warning in `self.lims_api.sample()` warns about the sample not being in LIMS. Resolving